libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -209,6 +223,11 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOpt
 /** Resync
  */
 - (SFSyncState*) reSync:(NSNumber*)syncId updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (syncId.integerValue == self.syncIdRunning) {
```

Comment:
But this is imposing an artificial constraint on the sync manager that doesn't otherwise exist: only one sync can run at a time.  That's a pretty drastic limitation to apps that manage many different sync data sets.